lib/embedded_ansible.rb

```diff
@@ -1,23 +1,156 @@
+require "securerandom"
+require "awesome_spawn"
+require "linux_admin"
+
 class EmbeddedAnsible
   APPLIANCE_ANSIBLE_DIRECTORY = "/opt/ansible-installer".freeze
   ANSIBLE_ROLE                = "embedded_ansible".freeze
+  SETUP_SCRIPT                = "#{APPLIANCE_ANSIBLE_DIRECTORY}/setup.sh".freeze
+  SECRET_KEY_FILE             = "/etc/tower/SECRET_KEY".freeze
+  CONFIGURE_EXCLUDE_TAGS      = "packages,migrations,firewall,supervisor".freeze
+  START_EXCLUDE_TAGS          = "packages,migrations,firewall".freeze
+  NGINX_HTTP_PORT             = "54321".freeze
+  NGINX_HTTPS_PORT            = "54322".freeze
 
   def self.available?
-    installed?
+    path = ENV["APPLIANCE_ANSIBLE_DIRECTORY"] || APPLIANCE_ANSIBLE_DIRECTORY
+    Dir.exist?(File.expand_path(path.to_s))
   end
 
   def self.enabled?
     MiqServer.my_server(true).has_active_role?(ANSIBLE_ROLE)
   end
 
   def self.running?
-    # TODO
-    false
+    services.all? { |service| LinuxAdmin::Service.new(service).running? }
   end
 
-  def self.installed?
-    path = ENV["APPLIANCE_ANSIBLE_DIRECTORY"] || APPLIANCE_ANSIBLE_DIRECTORY
-    Dir.exist?(File.expand_path(path.to_s))
+  def self.configured?
+    key = miq_database.ansible_secret_key
+    key.present? && key == File.read(SECRET_KEY_FILE)
+  end
+
+  def self.configure
+    configure_secret_key
+    run_setup_script(playbook_extra_variables.merge(:k => CONFIGURE_EXCLUDE_TAGS))
+    stop
+  end
+
+  def self.start
+    run_setup_script(playbook_extra_variables.merge(:k => START_EXCLUDE_TAGS))
+  end
+
+  def self.stop
+    services.each { |service| LinuxAdmin::Service.new(service).stop }
+  end
+
+  def self.disable
+    services.each { |service| LinuxAdmin::Service.new(service).stop.disable }
+  end
+
+  def self.services
+    AwesomeSpawn.run!("source /etc/sysconfig/ansible-tower; echo $TOWER_SERVICES").output.split
+  end
+
+  def self.playbook_extra_variables
+    json_value = {
+      :minimum_var_space => 0,
+      :nginx_http_port   => NGINX_HTTP_PORT,
+      :nginx_https_port  => NGINX_HTTPS_PORT
+    }.to_json
+    {:e => json_value}
+  end
+  private_class_method :playbook_extra_variables
+
+  def self.run_setup_script(params)
+    with_inventory_file do |inventory_file_path|
```

Comment:
Let's wrap this with a flock or something to avoid the possibility of the worker running more than one of these at once.